reservations/flightreservation/flightreservation.cpp

```diff
@@ -55,22 +61,61 @@ void FlightReservation::start()
 
 void FlightReservation::extract()
 {
-    QString reservationNumber = m_map["reservationNumber"].toString();
-    QString name = m_map["underName"].toMap().value("name").toString();
+    m_reservationNumber = m_map["reservationNumber"].toString();
+    m_name = m_map["underName"].toMap().value("name").toString();
     QVariantMap reservationForMap = m_map["reservationFor"].toMap();
 
     QString flightNameCode = reservationForMap["airline"].toMap().value("iataCode").toString();
     QString flightNumber = reservationForMap["flightNumber"].toString();
-    QString flight = flightNameCode + flightNumber;
+    m_flight = flightNameCode + flightNumber;
 
-    QString departureAirportName = reservationForMap["departureAirport"].toMap().value("name").toString();
-    QString departureAirportCode = reservationForMap["departureAirport"].toMap().value("iataCode").toString();
-    QDateTime departureTime = reservationForMap["departureTime"].toDateTime();
+    m_departureAirportName = reservationForMap["departureAirport"].toMap().value("name").toString();
+    m_departureAirportCode = reservationForMap["departureAirport"].toMap().value("iataCode").toString();
+    m_departureTime = reservationForMap["departureTime"].toDateTime();
 
-    QString arrivalAirportName = reservationForMap["arrivalAirport"].toMap().value("name").toString();
-    QString arrivalAirportCode = reservationForMap["arrivalAirport"].toMap().value("iataCode").toString();
-    QDateTime arrivalTime = reservationForMap["arrivalTime"].toDateTime();
+    m_arrivalAirportName = reservationForMap["arrivalAirport"].toMap().value("name").toString();
+    m_arrivalAirportCode = reservationForMap["arrivalAirport"].toMap().value("iataCode").toString();
+    m_arrivalTime = reservationForMap["arrivalTime"].toDateTime();
+
+    emit extractedData();
 }
 
-#include "flightreservation.moc"
+void FlightReservation::cacheData()
+{
+    QSqlDatabase db = QSqlDatabase::addDatabase("QSQLITE", "flight");
+    db.setDatabaseName("kdenowdb");
+
+    if (!db.open()) {
+        qWarning() << "Unable to open database";
+        qWarning() << db.lastError();
+    }
+    else {
+        qDebug() << "Database opened successfully";
+    }
+
+    QSqlQuery addQuery(db);
+    QString queryString = "create table if not exists Flight(id integer primary key autoincrement, reservationNumber varchar, name varchar, flight varchar, departureAirportName varchar, departureAirportCode varchar, departureTime varchar, arrivalAirportName varchar, arrivalAirportCode varchar, arrivalTime varchar)";
 
+    if (!addQuery.exec(queryString)) {
+        qWarning() << "Unable to create table";
+        qWarning() << addQuery.lastError();
+    }
+    else {
+        qDebug() << "Opened/Created successfully";
+    }
+
```

Comment:
Split code upto here into to a function initializeDatabase :  if (mQSqlDatabase == null) initializeDatabase();